### PR TITLE
RabbitMQ README / Message Bus Refactor Doc Update

### PR DESCRIPTION
**Pull request by kl5pierce**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Added sections to README, including:
- Pubsub vs. RPC communication for Shovel
- DataMover
- Backward compatibility with ZeroMQ (both directly by agent and multi-platform communication) 

Also made modifications to message bus refactor pages to include more information about SSL and minor edits for clarity. 

Please delete options that are not relevant.
- [X] This change requires a documentation update (change is a documentation update)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] Any dependent changes have been merged and published in downstream modules
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/shwethanidd%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/19616893%3Fv%3D4%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1875%23issuecomment-451576774%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222019-01-04T21%3A41%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/19616893%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shwethanidd%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20665ff24ea2a883f51329547192b6caba4533f69e%20examples/ListenerAgent/listener/agent.py%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1875%23discussion_r245177640%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Replace%20with%20%40PubSub.subscribe%28%27pubsub%27%2C%20%27%27%29%22%2C%20%22created_at%22%3A%20%222019-01-04T00%3A43%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/19616893%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shwethanidd%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20examples/ListenerAgent/listener/agent.py%3AL97-104%22%7D%2C%20%22Pull%20665ff24ea2a883f51329547192b6caba4533f69e%20docs/source/core_services/messagebus_refactor/RabbitMQ-Overview.rst%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1875%23discussion_r245178115%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22typo%2C%20unwanted%20text%20-%20%27isbind%20the%20queue%27.%20It%20should%20be%20only%20%27is%27%22%2C%20%22created_at%22%3A%20%222019-01-04T00%3A46%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/19616893%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shwethanidd%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20docs/source/core_services/messagebus_refactor/RabbitMQ-Overview.rst%3AL32-39%22%7D%2C%20%22Pull%20665ff24ea2a883f51329547192b6caba4533f69e%20volttron/utils/rmq_setup.py%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1875%23discussion_r245177953%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20is%20in%20correct.%20Copy%20contents%20of%20start_method%20from%20rabbitmq-volttron%20branch%22%2C%20%22created_at%22%3A%20%222019-01-04T00%3A45%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/19616893%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shwethanidd%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20volttron/utils/rmq_setup.py%3AL961-971%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/VOLTTRON/volttron/pull/1875%23issuecomment-451576774%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1875%23discussion_r245177640%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1875%23discussion_r245177953%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1875%23discussion_r245178115%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/shwethanidd'></a>

- [ ] <a href='#crh-comment-Pull 665ff24ea2a883f51329547192b6caba4533f69e examples/ListenerAgent/listener/agent.py 5'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1875#discussion_r245177640'>File: examples/ListenerAgent/listener/agent.py:L97-104</a></b>
- <a href='https://github.com/shwethanidd'></a> Replace with @PubSub.subscribe('pubsub', '')
- [ ] <a href='#crh-comment-Pull 665ff24ea2a883f51329547192b6caba4533f69e volttron/utils/rmq_setup.py 14'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1875#discussion_r245177953'>File: volttron/utils/rmq_setup.py:L961-971</a></b>
- <a href='https://github.com/shwethanidd'></a> This is in correct. Copy contents of start_method from rabbitmq-volttron branch
- [ ] <a href='#crh-comment-Pull 665ff24ea2a883f51329547192b6caba4533f69e docs/source/core_services/messagebus_refactor/RabbitMQ-Overview.rst 16'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1875#discussion_r245178115'>File: docs/source/core_services/messagebus_refactor/RabbitMQ-Overview.rst:L32-39</a></b>
- <a href='https://github.com/shwethanidd'></a> typo, unwanted text - 'isbind the queue'. It should be only 'is'

<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1875?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1875?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1875?approve=0'></a>&nbsp;<a href='https://github.com/VOLTTRON/volttron/pull/1875'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by craig8** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/VOLTTRON/volttron/pull/1875'>Refresh</a>.<a href='#crh-update-none'></a>

File: examples/ListenerAgent/listener/agent.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def onstart(self, sender, **kwargs):
             self.vip.heartbeat.start_with_period(self._heartbeat_period)
             self.vip.health.set_status(STATUS_GOOD, self._message)
 
-    @PubSub.subscribe('pubsub', '')
+    @PubSub.subscribe('pubsub', '', all_platforms=True)
```
Comment (shwethanidd) [inline examples/ListenerAgent/listener/agent.py]: Replace with @PubSub.subscribe('pubsub', '')

File: volttron/utils/rmq_setup.py [hunk 2]
```diff
@@ -961,14 +961,10 @@ def start_rabbit(rmq_home):
...
-                out, err = p.communicate()
-                rc = p.returncode
-                if rc != 0:
-                    _log.error("Error starting rabbitmq at {} Command out: {} "
-                               "Command err: {}".format(rmq_home, out, err))
-                    raise subprocess.CalledProcessError(cmd=start_cmd,
-                                                        returncode=rc)
+                
+                subprocess.check_call(start_cmd)
+                gevent.sleep(15)  # give a few seconds for all plugins to start
```
Comment (shwethanidd) [inline volttron/utils/rmq_setup.py]: This is in correct. Copy contents of start_method from rabbitmq-volttron branch

File: docs/source/core_services/messagebus_refactor/RabbitMQ-Overview.rst [hunk 3]
```diff
@@ -32,7 +32,7 @@ definitions that we need to know before we proceed.
 
     #. Direct - Here, the message is routed to a queue if the routing key of the message exactly matches the binding key of the queue.
 
-    #. Topic - Here, the message is routed to a queue based on pattern matching of the routing key with the binding key. The binding key and the routing key pattern must be a list of words delimited by dots, for example, "car.subaru.outback" or "car.subaru.*", "car.#". A message sent with a particular routing key will be delivered to all the queues that are bound with a matching binding key with some special rules as
+    #. Topic - Here, the message isbind the queue  routed to a queue based on pattern matching of the routing key with the binding key. The binding key and the routing key pattern must be a list of words delimited by dots, for example, "car.subaru.outback" or "car.subaru.*", "car.#". A message sent with a particular routing key will be delivered to all the queues that are bound with a matching binding key with some special rules as
```
Comment (shwethanidd) [inline docs/source/core_services/messagebus_refactor/RabbitMQ-Overview.rst]: typo, unwanted text - 'isbind the queue'. It should be only 'is'

**Comment by shwethanidd** [comment]
:shipit:

**Pull request merged**